src/contracts/LidoARM.sol

```diff
@@ -0,0 +1,117 @@
+    /// 10,000 = 100% performance fee
+    /// 1,500 = 15% performance fee
+    /// @param _feeCollector The account that can collect the performance fee
+    /// @param _capManager The address of the CapManager contract
+    function initialize(
+        string calldata _name,
+        string calldata _symbol,
+        address _operator,
+        uint256 _fee,
+        address _feeCollector,
+        address _capManager
+    ) external initializer {
+        _initARM(_operator, _name, _symbol, _fee, _feeCollector, _capManager);
+
+        // Approve the Lido withdrawal queue contract. Used for redemption requests.
+        steth.approve(address(lidoWithdrawalQueue), type(uint256).max);
+    }
+
+    /**
+     * @notice Request a stETH for ETH withdrawal.
+     * Reference: https://docs.lido.fi/contracts/withdrawal-queue-erc721/
+     * Note: There is a 1k amount limit. Caller should split large withdrawals in chunks of less or equal to 1k each.)
+     */
+    function requestLidoWithdrawals(uint256[] memory amounts)
+        external
+        onlyOperatorOrOwner
+        returns (uint256[] memory requestIds)
+    {
+        requestIds = lidoWithdrawalQueue.requestWithdrawals(amounts, address(this));
+
+        // Sum the total amount of stETH being withdraw
+        uint256 totalAmountRequested = 0;
+        for (uint256 i = 0; i < amounts.length; i++) {
+            totalAmountRequested += amounts[i];
+        }
+
+        // Increase the Ether outstanding from the Lido Withdrawal Queue
+        lidoWithdrawalQueueAmount += totalAmountRequested;
+    }
+
+    /**
+     * @notice Claim the ETH owed from the redemption requests and convert it to WETH.
+     * Before calling this method, caller should check on the request NFTs to ensure the withdrawal was processed.
+     */
+    function claimLidoWithdrawals(uint256[] memory requestIds) external onlyOperatorOrOwner {
+        uint256 etherBefore = address(this).balance;
+
+        // Claim the NFTs for ETH.
+        uint256 lastIndex = lidoWithdrawalQueue.getLastCheckpointIndex();
+        uint256[] memory hintIds = lidoWithdrawalQueue.findCheckpointHints(requestIds, 1, lastIndex);
+        lidoWithdrawalQueue.claimWithdrawals(requestIds, hintIds);
+
+        uint256 etherAfter = address(this).balance;
+
+        // Reduce the Ether outstanding from the Lido Withdrawal Queue
+        lidoWithdrawalQueueAmount -= etherAfter - etherBefore;
+
+        // Wrap all the received ETH to WETH.
+        weth.deposit{value: etherAfter}();
+    }
+
+    /**
+     * @dev Calculates the amount of stETH in the Lido Withdrawal Queue.
+     */
+    function _externalWithdrawQueue() internal view override returns (uint256) {
+        return lidoWithdrawalQueueAmount;
+    }
+
+    /// @notice This payable method is necessary for receiving ETH claimed from the Lido withdrawal queue.
+    receive() external payable {}
```

Comment:
🟣 ETH donations will get trapped on the contract. This fine. We could upgrade, no one one should be sending us ETH, and it won't affect our accounting. No change needed.

Reply:
We need to receive ETH from the Lido Withdrawal Queue in the `claimLidoWithdrawals` function. This is then converted to WETH.